lib/ex_doc/autolink.ex

```diff
@@ -92,17 +89,37 @@ defmodule ExDoc.Autolink do
   end
 
   defp custom_link(attrs, config) do
-    with {:ok, href} <- Keyword.fetch(attrs, :href),
-         [[_, text]] <- Regex.scan(~r/^`(.+)`$/, href) do
-      url(text, :custom_link, config)
-    else
-      _ -> nil
+    case Keyword.fetch(attrs, :href) do
+      {:ok, href} ->
+        case Regex.scan(~r/^`(.+)`$/, href) do
+          [[_, custom_link]] ->
+            build_custom_link(custom_link, config)
+
+          [] ->
+            build_extra_link(href, config)
+        end
+
+      _ ->
+        nil
+    end
+  end
+
+  defp build_custom_link(link, config) do
+    cond do
+      url = url(link, :custom_link, config) ->
+        url
+
+      true ->
+        uri = URI.parse(link)
+        message = "documentation references module \"#{uri.path}\" but it doesn't exist"
```

Comment:
saying _module_ in the warning is too generic, people could have written:

```
[bad](`Bad.bad/0`)
```

or even worse (please don't!)

```
[bad](`bad bad`)
```

and the warning would be incorrect. I'd just remove the word _module_ but hopefully we could use other parts that are already distinguishing between modules, functions, types, etc.

I also noticed that this test:

```
    test "bad" do
      assert_unchanged(~m"[bad](`String.upcase/9`)")
    end
```

warns twice:

```
warning: documentation references function String.upcase/9 but it is undefined or private
  nofile

warning: documentation references module "String.upcase/9" but it doesn't exist
  nofile
```

notice the first warning is more accurate so that's the one I'd rather keep here!

Reply:
what do you suggest for avoiding the double warning?

Reply:
from the two warnings we want to keep the 1st one, so let's not trigger the 2nd one at all. Now, we're not triggering the 1st under certain circumstances (e.g. `` `Unknown` ``, `` `Unknown.unknown/1` ``, etc) so I think we could have an option to _do_ trigger them when we have a custom link. WDYT?

Reply:
> from the two warnings we want to keep the 1st one, so let's not trigger the 2nd one at all. Now, we're not triggering the 1st under certain circumstances (e.g. `` `Unknown` ``, `` `Unknown.unknown/1` ``, etc) so I think we could have an option to _do_ trigger it when we have a custom link. WDYT?

please have a look